### PR TITLE
Add experimental SpacingAroundAngleBracketsRule

### PR DESCRIPTION
**Pull request by ymittal**
This builds on @christinalee's work in #587 (thanks so much, your PR was a perfect example of how to add new rules!)

In this PR, we created a lint rule to trim spacing around type parameters (`TYPE_PARAMETER_LIST`) and type arguments (`TYPE_ARGUMENT_LIST`).

Note that when the types span multiple lines, this rule does not fix the indentation itself. I rely on `IndentationRule` to dictate indentation. For example,
```
var map: Map<

                Int,
                List <   String >
    > = mapOf()
```
is formatted to
```
var map: Map<
                Int,
                List<String>
    > = mapOf()
```

The following shows that the indentation is fixed when all rules are run.
```
❯ cat sample | ./ktlint/build/run/ktlint --experimental -F --stdin
var map: Map<
    Int,
    List<String>
    > = mapOf()
```

### CODE REVIEW

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingAroundAngleBracketsRule.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        if (node.elementType.let { it == TYPE_PARAMETER_LIST || it == TYPE_ARGUMENT_LIST }) {
+            val openingBracket = node.firstChildNode
+            if (openingBracket != null) {
+                // Check for rogue spacing before an opening bracket:  Map <String, Int>
+                val beforeLeftAngle = openingBracket.prevLeaf()
+                if (beforeLeftAngle?.elementType == WHITE_SPACE) {
```
[the remark this answers is not in the document]
Comment (ymittal) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingAroundAngleBracketsRule.kt:27]: Good catch, yep you're right! I'm new to Kotlin - do you know if "fun" is the only keyword after which spacing is allowed?

[the remark this answers is not in the document]
Comment (romtsn) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingAroundAngleBracketsRule.kt:28]: Yep, I believe it also can be either `val` or `var`, like
```
inline val <T> T.exhaustive get() = this
```

[the remark this answers is not in the document]
Comment (romtsn) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingAroundAngleBracketsRule.kt:28]: And also just to make sure I would add test with `out`, `in` or `reified` keywords within the angle brackets (albeit this shouldn't change anything)
```
interface Consumer<in T> {
    fun add(item: T)
}
```

[the remark this answers is not in the document]
Comment (ymittal) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingAroundAngleBracketsRule.kt:29]: Thanks @romtsn! Added more tests

**Pull request merged**